core/include/kernel/msg_param.h

```diff
@@ -45,6 +45,19 @@ enum msg_param_mem_dir {
 };
 
 /**
+ * msg_param_mobj_from_noncontig_param() - construct mobj from non-contigous
+ * list of pages.
+ *
+ * @param - pointer to msg_param with OPTEE_MSG_ATTR_NONCONTIG flag set
+ * @bool  - true if buffer needs to be mapped into OP-TEE address space
+ *
+ * return:
+ *	mobj or NULL on error
+ */
+struct mobj *msg_param_mobj_from_noncontig_param(
```

Comment:
Any suggestions on how to rename it? 

We need `msg_param` prefix and we need to indicate that it constructs `mobj` from non-contiguous memory reference. 

(You know,  there are only two hard things in Computer Science: cache invalidation and naming things :) )